### PR TITLE
Fix TS compilation error.

### LINKED ISSUE #303: Typescript template does not work with start command
## 🐛 Bug Report

Simply running `npm run dev` of `npm run start` in a clean typescript template does not work:

image

Changing the import to `import { join } from "path"` fixes the issue but another one comes out:

image

## Your Environment

- *node version*: 12 and 14
- *fastify version*: >= 3
- *os*: Linux

Adding `esModuleInterop` to tsconfig.json fixes both.

### PR DESCRIPTION
**Pull request by kibertoad**
Also includes tests both for compilation and for trying to run the app from app-ts.

refs #303

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: templates/app-ts/src/app.ts [hunk 1]
```diff
@@ -18,17 +18,18 @@ const app: FastifyPluginAsync<AppOptions> = async (
...
   void fastify.register(AutoLoad, {
-    dir: path.join(__dirname, 'routes'),
+    dir: join(__dirname, 'routes'),
     options: opts
   })
 
 };
 
 export default app;
+export { app }
```
Comment (mcollina) [inline templates/app-ts/src/app.ts:35]: why this change?

Comment (kibertoad) [reply templates/app-ts/src/app.ts:35]: Fair point. This is my pet peeve with the lack of named exports to go with the default, so for convenience of writing a test I added it, but if you prefer having just a default, I can remove it.

Comment (mcollina) [reply templates/app-ts/src/app.ts:35]: Nice ok. Maybe in the test you can also check the default export?

Comment (kibertoad) [reply templates/app-ts/src/app.ts:35]: Done.

File: test/templates/app-ts.test.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import { fastify } from 'fastify'
+import { test } from 'tap'
+const sgetOriginal = require('simple-get').concat
+
+import appDefault, { app } from '../../templates/app-ts/src/app'
+
+const PORT = 3001;
```
Comment (mcollina) [inline test/templates/app-ts.test.ts:7]:
```suggestion
const PORT = 0;
```

Comment (mcollina) [reply test/templates/app-ts.test.ts:7]: use a random a picked port

Comment (kibertoad) [reply test/templates/app-ts.test.ts]: Done. Needed slight adjustment in test itself as well.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**